var/spack/repos/builtin/packages/scotch/package.py

```diff
@@ -171,20 +158,34 @@ def configure(self):
                 fh.write('\n'.join(makefile_inc))
 
     def install(self, spec, prefix):
-        self.configure()
-
         targets = ['scotch']
         if '+mpi' in self.spec:
             targets.append('ptscotch')
 
-        if '+esmumps' in self.spec:
-            targets.append('esmumps')
-            if '+mpi' in self.spec:
-                targets.append('ptesmumps')
+        if self.spec.version >= Version('6.0.0'):
```

Comment:
In 5.1.10b there is no target for esmumps or ptesmumps it is build by the targets scotch and ptscotch, that is why I removed the parallel build for ptscotch on version < 6.0.0
In fact I saw your comments on your commit that is why I tested the installation for all the possible combination of esmumps mpi and metis for version 6.0.0 and 5.1.10b but it is also possible that it works on my system due to a system dependency that you do not have on yours. So if you can at least test the scotch+esmumps@5.1.10b and let me know if it works for you.